docs/usage/self-hosted-experimental.md

```diff
@@ -0,0 +1,41 @@
+# Self-Hosted Experimental Options
+
+The following options are considered "experimental" in that:
+
+- They should not commonly be needed
+- They are typically an effort to work around some other service or platform's problem
+- We hope they can each be removed one day, or we are waiting to be sure they work as intended
+
+Experimental options which are commonly used and for which there is no external solution in sight should be replaced with official options.
+
+Please use experimental options at your own risk - they may be removed or behavior changed in any release, not just major, however all attempts will be made not to inconvenience anyone.
+
+## GITLAB_IGNORE_REPO_URL
+
+If set to any value, Renovate will ignore the Project's `http_url_to_repo` value and instead construct the git URL manually.
+
+## RENOVATE_CACHE_NPM_MINUTES
```

Comment:
We should have an admin option for all datasources. Currently  every datasource has it's own value 😕

Reply:
Yes agreed, it should be configurable and then this option not necessary.